sw/device/lib/crypto/drivers/entropy.c

```diff
@@ -104,6 +104,16 @@ static status_t csrng_send_app_cmd(uint32_t reg_address,
   for (size_t i = 0; i < cmd_len; ++i) {
     abs_mmio_write32(reg_address, cmd.seed_material->data[i]);
   }
+  // Wait for an "ack" signal before issuing new commands.
+  // TODO: How long does the CMD_STS bit stay asserted? Is there a chance we
+  // miss the ack here? Is it safe to skip the do/while and return an error if
+  // the ack bit isn't set?
+  bool ack = 0;
+  do {
+    reg = abs_mmio_read32(kBaseCsrng + CSRNG_SW_CMD_STS_REG_OFFSET);
+    ack = !bitfield_bit32_read(reg, CSRNG_SW_CMD_STS_CMD_STS_BIT);
```

Comment:
Could there be an issue with the `CSRNG_SW_CMD_STS_CMD_RDY_BIT` check above?

Reply:
You can probably refactor the initial `RDY_BIT` blocking statement and convert it into an error check, and then here you can poll for `RDY_BIT==0` and after done check `STS_BIT` to ensure there are no errors. wdyt?

@martin-lueker can weigh in as well.

Reply:
Lets go with the following approach:

1. The check at the beginning of this function can be refactored to the following:

```c
uint32_t reg = abs_mmio_read32(kBaseCsrng + CSRNG_SW_CMD_STS_REG_OFFSET);
if (!bitfield_bit32_read(reg, CSRNG_SW_CMD_STS_CMD_RDY_BIT)) {
  return // error //
}
```

2.a Update the ack check option A:

```c
do {
  reg = abs_mmio_read32(kBaseCsrng + CSRNG_SW_CMD_STS_REG_OFFSET);
  if (bitfield_bit32_read(reg, CSRNG_SW_CMD_STS_CMD_STS_BIT)) {
    return // error, detected request completed with an error //
  }
}  while (!bitfield_bit32_read(reg, CSRNG_SW_CMD_STS_CMD_RDY_BIT));
```

2.b Update the ack check option B:

```c
do {
  reg = abs_mmio_read32(kBaseCsrng + CSRNG_SW_CMD_STS_REG_OFFSET);
}  while (!bitfield_bit32_read(reg, CSRNG_SW_CMD_STS_CMD_RDY_BIT));

if (bitfield_bit32_read(reg, CSRNG_SW_CMD_STS_CMD_STS_BIT)) {
  return // error, detected request completed with an error //
}
```

Wether to use option 2.a versus 2.b depends on how the RTL was implemented.  @vogelpi  Is there someone can help us check this in the RTL side? Thanks!

Reply:
Based on the comments from @andreaskurth below. I believe 2.b should be good enough, but please let me know if you think otherwise. Thanks!

Reply:
Thanks for pinging me, as outline below I think neither of these proposals are safe. Because `RDY` will go high before the `STS` can be observed. As long as software pushes just a single command a time, `RDY` will be low just for a single clock cycle but `STS` may take multiple/many cycles to update.

Software should probably poll the interrupt status `cs_cmd_req_done` to decide when `STS` has been updated.